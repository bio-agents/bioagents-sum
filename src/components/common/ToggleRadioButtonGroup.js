import React from 'react'
import { Col, ControlLabel, FormGroup, ToggleButtonGroup } from 'react-bootstrap'

const ToggleRadioButtonGroup = ({label, valueChosen, children, input}) => (
  <FormGroup>
    <Col componentClass={ControlLabel} sm={2}>
      {label}
    </Col>
    <Col sm={10}>
      <ToggleButtonGroup type='radio' name={input.name} value={valueChosen} onChange={input.onChange}>
        {children}
      </ToggleButtonGroup>
    </Col>
  </FormGroup>
)

export default ToggleRadioButtonGroup
