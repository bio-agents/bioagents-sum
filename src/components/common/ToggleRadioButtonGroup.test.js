import React from 'react'
import renderer from 'react-test-renderer'
import ToggleRadioButtonGroup from './ToggleRadioButtonGroup'
import { ToggleButton } from 'react-bootstrap'

it('Renders ToggleRadioButtonGroup correctly', () => {
  const component = renderer.create(
    <ToggleRadioButtonGroup
      label='Toggle one thing, please'
      input={{ name: 'toggleRadioButtonGroup', onChange: value => value }}
      valueChosen='value1'
    >
      <ToggleButton value='value1'>{'Value 1'}</ToggleButton>
      <ToggleButton value='value2'>{'Value 2'}</ToggleButton>
      <ToggleButton value='value3'>{'Value 3'}</ToggleButton>
    </ToggleRadioButtonGroup>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
