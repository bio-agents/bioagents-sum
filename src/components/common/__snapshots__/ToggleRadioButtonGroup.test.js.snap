exports[`test Renders ToggleRadioButtonGroup correctly 1`] = `
<div
  className="form-group">
  <label
    className="col-sm-2 control-label"
    htmlFor={undefined}>
    Toggle one thing, please
  </label>
  <div
    className="col-sm-10">
    <div
      className="btn-group"
      data-toggle="buttons">
      <label
        className="btn btn-default active"
        disabled={false}
        type="button">
        <input
          autoComplete="off"
          checked={true}
          name="toggleRadioButtonGroup"
          onChange={[Function]}
          type="radio"
          value="value1" />
        Value 1
      </label>
      <label
        className="btn btn-default"
        disabled={false}
        type="button">
        <input
          autoComplete="off"
          checked={false}
          name="toggleRadioButtonGroup"
          onChange={[Function]}
          type="radio"
          value="value2" />
        Value 2
      </label>
      <label
        className="btn btn-default"
        disabled={false}
        type="button">
        <input
          autoComplete="off"
          checked={false}
          name="toggleRadioButtonGroup"
          onChange={[Function]}
          type="radio"
          value="value3" />
        Value 3
      </label>
    </div>
  </div>
</div>
`;
